src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -163,6 +167,33 @@ public String toString() {
         return groups.asUnmodifiableObservableMap();
     }
 
+    @Override
+    public ObservableList<PersonEvent> getUpcomingDates() {
+        List<PersonEvent> personEvents = new ArrayList<>();
+        persons.forEach(person -> {
+            // Add birthday
+            personEvents.add(new PersonEvent(person.getBirthday().getBirthday(), person,
+                    PersonEvent.getBirthdayDescription(person)));
+            // Add special dates
+            person.getDates().forEach(event -> personEvents.add(new PersonEvent(event.getDate(), person,
```

Comment:
So for here, it will just be `person.getDates().forEach(event -> personEvents.add(new PersonEvent(event.getDate(), person, event.toUi())`